docs/sources/next/javascript-api/jslib/utils/check.md

```diff
@@ -0,0 +1,62 @@
+---
+title: 'check( val, sets, [tags] )'
+description: 'Runs one or more checks on a value and generates a pass/fail result but does not throw errors or otherwise interrupt execution upon failure.'
+description: 'Runs one or more checks on a value and generates a pass/fail result but does not throw errors or otherwise interrupt execution upon failure.'
+---
+
+# check( val, sets, [tags] )
+
+This is a drop-in replacement for the built-in [check](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6/check/) with support for async values. Any `Promise`s will be awaited and the result will be reported once the operation has completed.
+
+| Parameter       | Type                   | Description                                                                              |
+| --------------- | ---------------------- | ---------------------------------------------------------------------------------------- |
+| val             | any                    | Value to test.                                                                           |
+| sets            | Record<string, any>    | Tests (checks) to run on the value. Functions will be called. `Promise`s will be awaited |
+| tags (optional) | Record<string, string> | Extra tags to attach to metrics emitted.                                                 |
+
+### Returns
+
+| Type             | Description |
+| ---------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Promise<boolean> | boolean     | `true` if all checks have succeeded, `false` otherwise. If any of the checked values was a `Promise`, the result will be wrapped in a `Promise`. |
```

Comment:
```suggestion
| Promise<boolean> | boolean     | `true` if all checks have succeeded, `false` otherwise. If any of the checked values was a `Promise`, the result is wrapped in a `Promise`. |
```